### PR TITLE
🧪 Update attribution tracking api to match latest spec

### PR DESCRIPTION
**Pull request by calebcordry**
Our code had gone out of sync with the latest proposal from chrome folks. See https://github.com/WICG/attribution-reporting-api/blob/main/EVENT.md#registering-attribution-sources

cc/ @zombifier

### CODE REVIEW

**Comment by calebcordry** [comment]
partial #35347

File: extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js [hunk 1]
```diff
@@ -379,9 +377,43 @@ describes.realWin(
...
+
+    it('should handle empty attributionsrc when given `browserAdConversion`', async () => {
+      env.sandbox
+        .stub(AmpAdExit.prototype, 'detectAttributionReportingSupport')
+        .returns(true);
+      const openStub = env.sandbox.stub(win, 'open').returns(win);
+      const config = {
+        targets: {
+          landingPage: {
+            finalUrl: 'https://advertiser.example',
```
Comment (sunsnap) [inline extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js:394]: When attributionsrc='', then finalUrl should be the full url starting with adtech domain.

File: extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js [hunk 2]
```diff
@@ -379,9 +377,43 @@ describes.realWin(
...
+
+      impl.executeAction({
+        method: 'exit',
+        args: {target: 'landingPage'},
+        event: makeClickEvent(1001),
+        satisfiesTrust: () => true,
+      });
+
+      expect(openStub).calledWithExactly(
+        'https://advertiser.example',
```
Comment (sunsnap) [inline extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js:414]: Same as above.

File: extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js]: Done.

File: extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-ad-exit/0.1/test/test-amp-ad-exit.js]: Done.

**Pull request merged**